crates/bitcoin-da/tests/bitcoin_da_tests.rs

```diff
@@ -386,47 +439,149 @@ mod tests {
         }
     }
 
+
     #[test]
-    fn test_read_height() {
+    fn test_read_data_by_block_height() {
+        // ===============================
+        // Given: a block with embedded data
+        // ===============================
+    
+        // Prepare the data and relayer configuration
+        let embedded_data = b"Hello, height!";
         let relayer = Relayer::new(&Config::new(
-            // "YOUR_NODE_IP:38332".to_owned(), // SIGNET
-            "localhost:8332".to_owned(), // REGNET
+            NODE_IP.to_owned(),
             "rpcuser".to_owned(),
             "rpcpass".to_owned(),
         ))
         .unwrap();
-        let height = 112; // change to whatever height that contains a tx
-        match relayer.read_height(height) {
+    
+        // Obtain the current block height before embedding
+        let blockchain_info = relayer.client.get_blockchain_info().unwrap();
+        let current_height = blockchain_info.blocks;
+    
+        // Embed the data into the blockchain by writing a transaction
+        match relayer.write(embedded_data) {
+            Ok(txid) => {
+                println!("Txid: {}", txid);
+                println!("Successful write");
+            }
+            Err(e) => panic!("Write failed with error: {:?}", e),
+        }
+    
+        // Add the transaction to a new block
+        relayer.generate_blocks(1).unwrap();
+    
+        // Ensure there's enough time for the transaction to be processed
+        std::thread::sleep(std::time::Duration::from_secs(2));
+    
+        // ================================
+        // When: read data by block height
+        // ================================
+    
+        let read_result = relayer.read_height(current_height + 1);
+    
+        // ==========================================
+        // Then: assert outcomes and expected results
+        // ==========================================
+    
+        // Assert the data was correctly embedded and can be read back
+        match read_result {
             Ok(data) => {
-                // Change this line to whatever data you want.
-                // "bark" is appended to the beginning of the data
-                assert_eq!(data, b"barkHello, world!".to_vec());
+                assert!(data.windows(b"barkHello, height!".len()).any(|window| window == b"barkHello, height!"));
                 println!("Successful read");
             }
             Err(e) => panic!("Read failed with error: {:?}", e),
         }
     }
+    
+    #[cfg(feature = "regnet")]
+    #[test]
+    fn test_read_transaction_from_mempool() {
```

Comment:
Here, you added two tests to read the data from a txid in the mempool. 
The test we had was to read the data from a txid AND a block height, we should still have it somewhere.

Good tests tho

Reply:
Placed them back in

Reply:
Placed them back in